src/Microsoft.Dnx.Runtime/LockFileBasedProjectResolver.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Dnx.Runtime
+{
+    public class LockFileBasedProjectResolver : IProjectResolver
```

Comment:
This implementation just needs to create the dictionary up front mapping names to paths and nothing more. Do the computation in the ctor.